src/Common/src/System/CharArrayHelpers.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System
+{
+    internal static class CharArrayHelpers
+    {
+        internal static bool EqualsOrdinal(string left, char[] right, int rightStartIndex, int rightLength)
+        {
+            Debug.Assert(left != null);
+            DebugAssertArrayInputs(right, rightStartIndex, rightLength);
+
+            if (left.Length != rightLength)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                if (left[i] != right[rightStartIndex + i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        internal static bool EqualsOrdinalAsciiIgnoreCase(string left, char[] right, int rightStartIndex, int rightLength)
+        {
+            Debug.Assert(left != null);
+            DebugAssertArrayInputs(right, rightStartIndex, rightLength);
+
+            if (left.Length != rightLength)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                uint charA = left[i];
+                uint charB = right[rightStartIndex + i];
+
+                // We're only interested in ASCII characters here.
+                if ((charA - 'a') <= ('z' - 'a')) charA -= 0x20;
+                if ((charB - 'a') <= ('z' - 'a')) charB -= 0x20;
+
+                if (charA != charB)
```

Comment:
Sure, I can fix. BTW, I just lifted this from CompareInfo: https://github.com/dotnet/coreclr/blob/43b39a73cbf832ec13ec29ed356cb75834e7a8d7/src/mscorlib/corefx/System/Globalization/CompareInfo.cs#L319-L320